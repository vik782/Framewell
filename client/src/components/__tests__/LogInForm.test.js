import React from "react";
import LogInForm from "../src/pages/Login/LoginForm";
import { render, fireEvent, getByTestId } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

// Snapshot test to account for changes to the UI, whether expected or unexpected
test("Login Form renders correctly", () => {
    const loginForm = renderer.create(<MemoryRouter><LogInForm /></MemoryRouter>).toJSON();
    expect(loginForm).toMatchSnapshot();
});

// Unit Testing
test("Username and Password input field initially empty", () => {
    const { getByTestId } = render(<MemoryRouter><LogInForm /></MemoryRouter>);
    const usernameField = getByTestId("userField");
    const passwordField = getByTestId("passwordField");

    expect(usernameField.textContent).toBe("");
    expect(passwordField.textContent).toBe("");
});

test("Username input field changes correctly", () => {
    const { getByTestId } = render(<MemoryRouter><LogInForm /></MemoryRouter>);
    const usernameField = getByTestId("userField");
    const passwordField = getByTestId("passwordField");

    fireEvent.change(usernameField, {
        target: {
            value: "username"
        }
    });

    fireEvent.change(passwordField, {
        target: {
            value: "password"
        }
    });

    expect(usernameField.value).toBe("username");
    expect(passwordField.value).toBe("password");
});

test("Sign In Button submits form", () => {
    
});

test("Returned feedback is correct", () => {
    
});

