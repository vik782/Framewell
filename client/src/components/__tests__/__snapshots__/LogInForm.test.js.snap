// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Form renders correctly 1`] = `
<form
  action="/login"
  data-testid="submit-form"
  method="post"
  onSubmit={[Function]}
>
  <ul>
    <li>
      <label>
         Username 
      </label>
    </li>
    <li
      className="input-field login-initial"
    >
      <span>
        <span />
      </span>
      <input
        data-testid="userField"
        id="userName"
        onChange={[Function]}
        type="text"
      />
    </li>
    <li>
      <label>
         Password 
      </label>
    </li>
    <li
      className="input-field login-initial"
    >
      <span>
        <span />
      </span>
      <input
        data-testid="passwordField"
        id="password"
        onChange={[Function]}
        type="password"
      />
      <span>
        <span />
      </span>
    </li>
    <li
      className="forget-password-link"
    >
      <h5>
        Forgot password?
      </h5>
    </li>
    <li>
      <p
        className="feedback"
        data-testid="feedback-test"
      />
    </li>
    <li>
      <button
        data-testid="submit-btn"
        type="submit"
      >
        Sign In
      </button>
    </li>
  </ul>
</form>
`;
